source/agora/node/Registry.d

```diff
@@ -711,6 +708,13 @@ private struct TypedPayload
             assert(this.payload.addresses.length == 1);
             dg(ResourceRecord.make!(TYPE.CNAME)(name, this.payload.ttl,
                 Domain.fromString(this.payload.addresses[0].host)));
+
+            // Add URI record along with
+            dg(ResourceRecord.make!(TYPE.URI)(
+                Domain.fromString(format(uri_service_label, name.toString)),
```

Comment:
Note: You do not need to call `toString` here, `format` will do that for you.
In fact, it should call the `toString` overload with a `sink`, reducing the amount of allocations.

Reply:
Another easy improvement: We know that an address can only be 256 chars long at most. So we could format this into a static buffer because it can never exceed 256 chars (something we do not check BTW, and could cause problems).

Reply:
> (something we do not check BTW, and could cause problems).

`Domain.fromString` has `ensure` for that

Reply:
> So we could format this into a static buffer because it can never exceed 256 chars

`Domain.fromString` checks `length` of the array not `strlen`, thus I think static array would cause problems there.

Reply:
```D
            ensure(uri_service_label.length + name.toString <= Domain.MaxLength, "Useful msg");
            char[Domain.MaxLength] buffer;
            dg(ResourceRecord.make!(TYPE.URI)(
                Domain.fromString(snformat(buffer, uri_service_label, name)),
```

This would not allocate, and work as expected. It does not allocate because Ocean's `snformat` will write to a static array up to its length, and it returns a slice to the subset of the array that has been written to.
I am not saying we should use it here - There are plenty of memory optimizations needed and I don't think we are there yet. But I wanted to share the pattern.